### ISSUE TITLE
ENH: Expose parallel HiGHS solvers in high-level API

### ISSUE THREAD

**Issue by fuglede** [scipy/scipy#17250]
### Is your feature request related to a problem? Please describe.

Currently, it's possible to [provide an option called `'parallel'`](https://github.com/scipy/scipy/blob/main/scipy/optimize/_highs/cython/src/_highs_wrapper.pyx#L369) to `scipy.optimize._highs._highs_wrapper._highs_wrapper`. Setting this to `True`, while I haven't been able to find a case where my solution times improve, it does cause multiple CPUs to spin, so it does do something.

The functionality doesn't survive through any of the higher level APIs though; the only way I've been able to make use of it without recompiling the Cython is by manually adding a parameter called `parallel` to `scipy.optimize._linprog_highs._linprog_highs` and making sure to pass it through to `_highs_wrapper`; indeed, `_linprog_highs` constructs its own dictionary of options and doesn't include any additional options one might provide it with.

Similarly, `_highs_wrapper` can take options called `min_threads` and `max_threads`. I tried a handful of different combinations of the two options but no matter what I pass, Python crashes.

**Comment by mdhaber** [comment]
I suppose we can relax `linprog` to pass unrecognized options through. In the meantime, that is what [`milp`](https://scipy.github.io/devdocs/reference/generated/scipy.optimize.milp.html) does, so you should be able to use that.

**Comment by AbhishekVermaDH** [comment]
Highs solver interface has been a big selling point for us to recently consider scipy interface. Thanks a lot for making this happen.

Many OR practitioner honor their problem types (LP/MILP) like programmers honor their class name/types. So it's very likely they will restrict themselves to `linprog` for LPs; ignoring `milp` (thinking that's only for MILPs). This may leave bitter taste in users.

Relaxing `linprog` is very simple and AFAIK mostly harmless. So if we want to be appealing to OR community who cares about optimality and speed, then this change will be a big plus.

**Comment by mdhaber** [comment]
PR welcome!

**Comment by mdhaber** [comment]
@fuglede If it sounds good to you, I'll let gh-17961 resolve this issue.

**Comment by fuglede** [comment]
Yep, I didn't test it, but it looks it would do what it should, and I imagine the change to `threads` vs. `max_threads` will take care of the segfaults(?)

In terms of providing a test case where the parallel solver gives a (wall time) improvement: If I understand correctly, everything is passed through verbatim so time-saving examples are exactly those that would be examples when using HiGHS directly. So maybe @jajhall (hope you don't mind the ping!) has a favorite example we could use?
